src/react/components/DealOverview.tsx

```diff
@@ -99,16 +101,37 @@ export const DealOverview = () => {
 	}, [repaymentSelectValue, deal]);
 
 	useEffect(() => {
-		if (wallet?.publicKey && connection.connection) {
-			setPlaceholder("0");
-		} else {
-			setPlaceholder("Connect wallet");
+		const dealNumber = params.deal && parseInt(params.deal) - 1;
+
+		if ((!dealNumber && dealNumber !== 0) || dealNumber < 0) {
+			navigate(Path.NOT_FOUND);
+			return;
 		}
 
+		setDealNumber(dealNumber);
+	}, [params.deal, wallet, connection.connection, navigate]);
+
+	useEffect(() => {
+		try {
+			setBorrower(new PublicKey(params.borrower || ""));
+		} catch (e) {
+			navigate(Path.NOT_FOUND);
+		}
+	}, [params.borrower, navigate]);
+
+	useEffect(() => {
 		if (wallet?.publicKey && connection.connection) {
 			fetchDealData();
 		}
-	}, [connection.connection, wallet?.publicKey, fetchDealData]);
+	}, [wallet, connection.connection, fetchDealData]);
+
+	useEffect(() => {
+		if (wallet?.publicKey && connection.connection) {
+			setPlaceholder("0");
```

Comment:
maybe another placeholder for the key? `xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx` or something?

Reply:
there is a ticket for this

Reply:
we discussed it this morning during the call and a ticket was made

Reply:
indeed, didn't know there was already a ticket 👍